tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```diff
@@ -1,12 +1,97 @@
 package com.linagora.tmail.webadmin.cleanup;
 
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.DefaultMailboxes;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.store.StoreMailboxManager;
 import org.apache.james.task.Task;
+import org.apache.james.user.api.UsersRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import com.linagora.tmail.james.jmap.settings.JmapSettings;
+import com.linagora.tmail.james.jmap.settings.JmapSettingsRepository;
+
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class CleanupTrashService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CleanupTrashService.class);
+    private static final String WEEKLY_PERIOD = "weekly";
+    private static final String MONTHLY_PERIOD = "monthly";
+    private static final String TRASH_CLEANUP_ENABLED_SETTING = "trash.cleanup.enabled";
+    private static final String TRASH_CLEANUP_PERIOD_SETTING = "trash.cleanup.period";
+    private final UsersRepository usersRepository;
+    private final JmapSettingsRepository jmapSettingsRepository;
+    private final StoreMailboxManager storeMailboxManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public CleanupTrashService(UsersRepository usersRepository, JmapSettingsRepository jmapSettingsRepository, StoreMailboxManager storeMailboxManager, SessionProvider sessionProvider) {
+        this.usersRepository = usersRepository;
+        this.jmapSettingsRepository = jmapSettingsRepository;
+        this.storeMailboxManager = storeMailboxManager;
+        this.sessionProvider = sessionProvider;
+    }
 
     public Mono<Task.Result> cleanupTrash() {
-        return Mono.just(Task.Result.COMPLETED);
+        return Flux.from(usersRepository.listReactive())
+            .flatMap(this::checkSettingAndCleanupTrashForSingleUser)
```

Comment:
IMO we should put lower concurrency for this flatMap. Otherwise, we would use default concurrency 128 to get 128 users' messages at the same time -> a lot of pressure on Cassandra.